test/test_util.py

```diff
@@ -455,7 +458,33 @@ def test_image_decoder_b64(self):
 
     @patch("weaviate.util.uuid_lib")
     def test_generate_uuid5(self, mock_uuid):
-
         result = generate_uuid5("TestID!", "Test!")
         self.assertIsInstance(result, str)
         mock_uuid.uuid5.assert_called()
+
+    def test_parse_version_string(self):
+        assert parse_version_string("v1.18.1") == parse_version_string("1.18.1")
+        assert parse_version_string("v1.18.1") == parse_version_string("1.18.5")
+        assert len(parse_version_string("v1.18.1")) == 2
+        assert len(parse_version_string("1.1.5")) == 2
+        assert len(parse_version_string("2")) == 2
+        assert parse_version_string("v1.18.3") == (1, 18)
+        assert parse_version_string("1.1.5") == (1, 1)
+        assert parse_version_string("2") == (2, 0)
+        assert parse_version_string("2.0.0") == (2, 0)
+        assert parse_version_string("v2.05.5") == (2, 5)
+
+
+@pytest.mark.parametrize(
+    "version,too_old",
+    [
+        ("v1.0.0", True),
+        ("v1.10.5", True),
+        ("v1.15.3", True),
+        ("v1.18.3", False),
+        ("v1.25.0", False),
+        ("v2.0.0", False),
+    ],
```

Comment:
One problem with this is that we need to change the test.

Could we define MINIMUM_NO_WARNING_VERSION_MAJOR and MINIMUM_NO_WARNING_VERSION_MINOR instead of MINIMUM_NO_WARNING_VERSION and then construct the tests out of them?

Reply:
Ah yes. What do you think about this? This way `MINIMUM_NO_WARNING_VERSION` can be defined in the same format as what the server returns.

```
MINIMUM_NO_WARNING_VERSION_MAJOR, MINIMUM_NO_WARNING_VERSION_MINOR = parse_version_string(MINIMUM_NO_WARNING_VERSION)
MINIMUM_NO_WARNING_VERSION_MAJOR = int(MINIMUM_NO_WARNING_VERSION_MAJOR)
MINIMUM_NO_WARNING_VERSION_MINOR = int(MINIMUM_NO_WARNING_VERSION_MINOR)
@pytest.mark.parametrize(
    "version,too_old",
    [
        (f"v{MINIMUM_NO_WARNING_VERSION_MAJOR-1}.{MINIMUM_NO_WARNING_VERSION_MINOR-1}.1", True),
        (f"v{MINIMUM_NO_WARNING_VERSION_MAJOR-1}.{MINIMUM_NO_WARNING_VERSION_MINOR+1}.2", True),
        (f"v{MINIMUM_NO_WARNING_VERSION_MAJOR}.{MINIMUM_NO_WARNING_VERSION_MINOR-1}.3", True),
        (MINIMUM_NO_WARNING_VERSION, False),
        (f"v{MINIMUM_NO_WARNING_VERSION_MAJOR}.{MINIMUM_NO_WARNING_VERSION_MINOR+1}.0", False),
        (f"v{MINIMUM_NO_WARNING_VERSION_MAJOR+1}.{MINIMUM_NO_WARNING_VERSION_MINOR-1}.0", False),
        (f"v{MINIMUM_NO_WARNING_VERSION_MAJOR+1}.{MINIMUM_NO_WARNING_VERSION_MINOR+1}.0", False),
    ],
)
def test_is_weaviate_too_old(version: str, too_old: bool):

    assert is_weaviate_too_old(version) is too_old
```

Reply:
that would also work :)

Reply:
thanks!